### PR TITLE
Compiler warning fixes

### PR DESCRIPTION
**Pull request by permcody**
Cleanup patch. Some of the variables look important. I wouldn't mind extra sets of eyes on this.
@sveerara, @aeslaughter, @bwspenc

### CODE REVIEW

**Comment by permcody** [comment]
Oh and @jiangwen84

File: modules/solid_mechanics/src/materials/IsotropicPowerLawHardening.C [hunk 1]
```diff
@@ -65,7 +65,7 @@ IsotropicPowerLawHardening::computeStressInitialize(unsigned qp, Real effectiveT
 }
 
 Real
-IsotropicPowerLawHardening::computeHardening( unsigned qp, Real scalar )
+IsotropicPowerLawHardening::computeHardening( unsigned /*qp*/, Real scalar )
 {
   Real stress = _effectiveTrialStress - 3.0 * _shear_modulus * scalar;
   Real slope = std::pow(stress, 1.0/_n -1.0)/ _n/ std::pow(_K, 1.0/_n) - 1.0/_youngs_modulus;
```
Comment (sveerara) [inline modules/solid_mechanics/src/materials/IsotropicPowerLawHardening.C]: The changes to IsotropicPowerLawHardening look fine. The yield stress and the hardening slope do not depend on qp in this case. Thanks!

File: modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentGSS.C [hunk 2]
```diff
@@ -34,7 +34,7 @@ CrystalPlasticityStateVarRateComponentGSS::calcStateVariableEvolutionRateCompone
 
   Real r = _hprops[0];
   Real h0 = _hprops[1];
-  Real tau_init = _hprops[2];
+//  Real tau_init = _hprops[2];
```
Comment (jiangwen84) [inline modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentGSS.C]: we can just delete this variable as it is not being used at all. Thanks!

Comment (permcody) [inline modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentGSS.C]: Well my point is: Why is it not being used? It's in the middle of this vector and you're just ignoring it? Is this a design issue?

Comment (jiangwen84) [inline modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentGSS.C]: That is a good point. I think I began to use this variable for an old formulation and now I do not need this variable. You can leave it as it is now and I will change this as well as some affected tests files by a new PR. Thanks!

**Pull request merged**